lxd/device/device_utils_network.go

```diff
@@ -516,3 +518,81 @@ func networkValidVLANList(value string) error {
 
 	return nil
 }
+
+// bgpAddPrefix adds external routes to the BGP server.
+func bgpAddPrefix(d *deviceCommon, config map[string]string) error {
+	// BGP is only valid when tied to a managed network.
+	if config["network"] == "" {
+		return nil
+	}
+
+	// Load the network configuration.
+	n, err := network.LoadByName(d.state, project.Default, config["network"])
```

Comment:
I've noticed that in `bridged` NIC there are a few places where the parent network is loaded.

In the `ovn` NIC I avoided the need for loading the network multiple times by validating the network/parent and then loading it in the `validateConfig` function and then storing it inside the device as `d.network`.

```go
type nicOVN struct {
	deviceCommon

	network ovnNet // Populated in validateConfig().
}
```

https://github.com/lxc/lxd/blob/master/lxd/device/nic_ovn.go#L132

I propose we add this same approach to the NIC types that also need to access the network's config after validation.

And then this function can be updated to accept the loaded network as an argument to avoid loading it again.

The `bridged` NIC driver can then have the loader removed from `setFilters()` and `State()`.